colpali_engine/models/qwen2/colqwen2/processing_colqwen2.py

```diff
@@ -9,11 +9,36 @@
 from colpali_engine.utils.processing_utils import BaseVisualRetrieverProcessor
 
 
+def round_by_factor(number: float, factor: int) -> int:
+    """Returns the closest integer to 'number' that is divisible by 'factor'."""
+    return round(number / factor) * factor
+
+
+def ceil_by_factor(number: float, factor: int) -> int:
+    """Returns the smallest integer greater than or equal to 'number' that is divisible by 'factor'."""
+    return math.ceil(number / factor) * factor
+
+
+def floor_by_factor(number: float, factor: int) -> int:
+    """Returns the largest integer less than or equal to 'number' that is divisible by 'factor'."""
+    return math.floor(number / factor) * factor
+
+
 class ColQwen2Processor(BaseVisualRetrieverProcessor, Qwen2VLProcessor):
     """
     Processor for ColQwen2.
     """
 
+    visual_prompt_prefix: ClassVar[str] = (
+        "<|im_start|>user\n<|vision_start|><|image_pad|><|vision_end|>Describe the image.<|im_end|>\n"
+    )
+    query_augmentation_token: ClassVar[str] = "<pad>"
+    image_token: ClassVar[str] = "<|image_pad|>"
```

Comment:
this could be the non-string version as well ? but doesn't really matter right now

Reply:
As I had noticed yesterday, it seems that the `<|image_pad|>` cannot be accessed properly using the current transformers implementation. I'll keep it like this for now.